go/vt/vtgate/engine/update.go

```diff
@@ -232,16 +230,6 @@ func (upd *Update) updateVindexEntries(vcursor VCursor, bindVars map[string]*que
 		for _, colVindex := range upd.Table.Owned {
 			// Update columns only if they're being changed.
 			if updColValues, ok := upd.ChangedVindexValues[colVindex.Name]; ok {
-				offset := updColValues.Offset
-				if !row[offset].IsNull() {
-					val, err := evalengine.ToInt64(row[offset])
-					if err != nil {
-						return err
-					}
-					if val == int64(1) { // 1 means that the old and new value are same and vindex update is not required.
-						continue
-					}
-				}
```

Comment:
This snippet started fails the following test case: `TestUpdateEqualWithWriteOnlyLookupUniqueVindex`. We were checking if the new value we want to assign is equal to `int64(1)` in which case we continue and go to the next updated value. This logic is incorrect when the vindex's original value is equal to `1`. The test `TestUpdateEqualWithWriteOnlyLookupUniqueVindex` represents this issue by having the `lu_col` vindex originally set to `1` and updating it to `5`.

Reply:
The right-most select expressions of the initial update statement are here to tell us whether we are updating a column to the same value or not. The value of those right-most fields is either `0` or `1` (`0` if updating to a different value, `1` if updating to the same value). The snippet deletion will be reverted, resolving this conversation.